packages/design-system-react-native/src/primitives/AvatarBase/AvatarBase.types.ts

```diff
@@ -38,6 +38,18 @@ export type AvatarBaseProps = {
    * fails to render
    */
   fallbackTextProps?: Omit<TextProps, 'children'>;
+  /**
+   * Optional prop to include the border with the Avatar.
+   * For internal use only
+   * @default false
+   */
+  hasBorder?: boolean;
+  /**
+   * Optional prop to make sure there's no transparency in the background color
+   * For internal use only
+   * @default false
+   */
+  hasSolidBackgroundColor?: boolean;
```

Comment:
These are needed for `AvatarGroup` and `Badge`

Reply:
non-blocking: Can we create a ticket or separate PR to add these props to the React component to ensure alignment across our component APIs? I think it's fine to handle the styling logic differently internally within the component.